library/core/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1931,4 +1947,68 @@ private static boolean parseMsAcmCodecPrivate(ParsableByteArray buffer) throws P
 
   }
 
+  private class TrueHDSampleHolder {
+
+    private int counter;
+    private int samplesSize;
+    private long timeUs;
+    private @C.BufferFlags int blockFlags;
+    private boolean gotSyncFrame;
+    private boolean letFirstIn;
+    private Track track;
+
+    TrueHDSampleHolder() {
+      reset();
+    }
+
+    private boolean isFirstSampleCheck() {
+      return counter == 1;
+    }
+
+    private void resetCounters() {
+      track = null;
+      counter = 0;
+      samplesSize = 0;
+      timeUs = 0;
+      blockFlags = 0;
+    }
+
+    void reset() {
+      gotSyncFrame = false;
+      letFirstIn = true;
+      resetCounters();
+    }
+
+    void testSample(ExtractorInput input, int size, @C.BufferFlags int blockFlags, Track track)
+            throws IOException, InterruptedException {
+      samplesSize += size;
+      ++counter;
+      if (isFirstSampleCheck())
+        this.blockFlags = blockFlags;
+      if (!gotSyncFrame) {
+        byte[] bytes = new byte[12];
+        input.peekFully(bytes, 0, 12);
```

Comment:
This extractor supports header stripping compression where some specified bytes are prepended to every output sample. I'm not sure it makes sense to use that with TrueHD audio tracks, given how the samples are fragmented. If it were possible, we'd need to read from sampleStrippedBytes first (see readToTarget). If it's definitely not possible, maybe worth asserting that track.sampleStrippedBytes is null.

Reply:
I'll put in the code to check sampleStrippedBytes if it has data, but I don't think I have anything to actually test if that case were to ever happen.